test.js

```diff
@@ -101,20 +115,47 @@ test('toVFile.read', function (t) {
     });
   });
 
-  t.test('should pass an error on non-existing files', function (st) {
-    st.plan(2);
+  t.test('should work in promise mode (string with encoding)', function (st) {
+    st.plan(3);
+
+    vfile.read('readme.md', 'utf8')
+      .then(function (file) {
+        st.equal(file.path, 'readme.md');
+        st.equal(typeof file.contents, 'string');
+        st.equal(file.toString(), fixture);
+      })
+      .catch(function () {
+        st.fail('should resolve, not reject');
+      });
+  });
+
+  t.test('should return an error on non-existing files', function (st) {
+    st.plan(3);
 
     vfile.read('missing.md', 'utf8', function (err, file) {
       st.equal(file, undefined);
+      st.ok(err instanceof Error);
       st.ok(/ENOENT/.test(err.message));
     });
   });
 
-  t.end();
+  t.test('should reject on non-existing files in promise mode', function (st) {
+    st.plan(2);
+
+    vfile.read('missing.md')
+      .then(function () {
+        st.fail('should reject, not resolve');
+      })
+      .catch(function (err) {
+        st.ok(err instanceof Error);
+        st.ok(/ENOENT/.test(err.message));
+      });
+  });
 });
 
 test('toVFile.writeSync', function (t) {
   var filePath = 'fixture.txt';
+  var invalidFilePath = 'invalid/path/to/fixture.txt';
```

Comment:
should this use `path.join` to support people using windows?

Reply:
This is a relative path, which should work on Windows too.

Reply:
Do forward slashes work on Windows 🤔 I though, whether relative or not, they’re always forward slashes (except in the Windows Subshell)

Reply:
I haven't used Windows for quite some time, but from what I remember, forward slashes are just as OK as backslashes in paths there. There is a machine next to me with Windows – I can try running tests there now if I manage to install nodejs 😆

Reply:
No luck! That machine did not have a mouse and was not connected to LAN – can't even login 😅

Reply:
Lol! Hmm, maybe you’re right. If you can test this and it works, I’m okay with it. But I’d rather have `path.join('foo', 'bar', 'baz')`. I think that should work everywhere!

Could you do that? Or provide results back from Windows?

Reply:
Thanks for trying btw! 😉

Reply:
Nailed it on another machine! Unfortunately, two tests failed, but that is unrelated to the new ones:

![failures](https://user-images.githubusercontent.com/608862/35877177-8762e5f6-0b6c-11e8-9b9a-d1905f06cf63.PNG)

![error-source](https://user-images.githubusercontent.com/608862/35877182-8d62a356-0b6c-11e8-846a-4a03c5e73750.PNG)

I'm afraid I have to set off for now. WDYT of setting up Travis and Appveyor CI to catch similar errors in future without having to go through the Windows hell? Perhaps you could find inspiration in https://github.com/Glavin001/atom-beautify, where all PRs have their tests automatically ran on *nix and Windows. Useful stuff!

Oh gosh _I've seen things_... 😆 
![crlf](https://user-images.githubusercontent.com/608862/35877229-addbbeba-0b6c-11e8-8daa-be1de058cbca.PNG)

Reply:
Lol! That last one is horrid! Yeah, maybe we should just add windows tests as well.
But that should be another PR! This looks good for now!

Reply:
Great! I’ll be happy if you cut a new release - keen to use promisified to-vfile in one experiment!

Reply:
Alright, will do!